packages/components/src/custom-select-control-v2/types.ts

```diff
@@ -49,6 +55,42 @@ export type CustomSelectProps = {
 	value?: string | string[];
 };
 
+export type Option =
+	| {
+			key: string;
+			name: string;
+			style?: {};
+			className?: string;
+			__experimentalHint?: string;
+	  }
+	| { selectedItem: string | string[] };
```

Comment:
This is a bit a messy - it seems to be the best way to translate the old option to the new value, but I am sure it can be typed better.

Reply:
It can be! 😄 Updated in https://github.com/WordPress/gutenberg/pull/57000/commits/ee89e01af3af79b8d718483dde101733697bd458 after looking with @ciampo